### ISSUE TITLE
UUID migrations issue

### ISSUE THREAD

**Issue by fmonts** [doctrine/dbal#2438]
I have the following situation:

    class Site
    {
        /**
         * @ORM\Id
         * @ORM\GeneratedValue(strategy="AUTO")
         * @ORM\Column(type="integer")
         */
        private $id;
    

        //other fields...
    }

And I want to add a uuid (I don't want to expose the real id, for obvious reasons). So I added:

    /**
     * @ORM\Column
     * @ORM\GeneratedValue(strategy="UUID")
     */
    private $uuid;

The odd thing is that when I run `doctrine:schema:update` or `doctrine:migrations:diff`, I get the following output:

    ALTER TABLE sites ADD uuid VARCHAR(255) NOT NULL, CHANGE id id INT NOT NULL;

In other words, the existing `id` column loses its AUTO_INCREMENT, and that breaks a lot of existing relations with other entities...
(also but lesser important, VARCHAR(255) doesn't seem the best choice for such a fixed length value)

The expected behaviour is:
     - add a unique key on the uuid, and leave the AI on the id

I'm using doctrine 2.5.4, mysql 5.7.13, symfony 3.1.2, php 7.0.8

**Comment by deeky666** [comment]
`@ORM\GeneratedValue` is only intended for a single primary key column and not for regular columns ([according to the documentation](http://doctrine-orm.readthedocs.io/projects/doctrine-orm/en/latest/reference/annotations-reference.html#annref-generatedvalue)). The fact that the schema tool maps that column to `VARCHAR(255)` is because you are not specifying a column type. You should use `@ORM\Column(type="guid")` instead. The fact that your primary key column looses the `AUTOINCREMENT` property is most probably because of the misuse of two `@ORM\GeneratedValue` annotations.

So what you need to do is decide on whether your `id` or your `uuid` should be the primary key and use the appropriate generator strategy there.

**Comment by deeky666** [comment]
Closing as this is not an issue IMO.

**Comment by fmonts** [comment]
@deeky666 thank you for your reply. 
Is there any way to have a guid primary key, and another auto_increment field?

**Comment by deeky666** [comment]
@fmonts I don't think so because you can only have one column with a generated value (at least according to the documentation). DBAL wise it should be possible I think but I am not sure about ORM.

**Comment by Ocramius** [comment]
In your migration, you'd have to:

 1. add a nullable, unique `GUID` column
 2. run an `UPDATE` statement on the entire table, setting a new `GUID` for every record. This step depends on the DB that you use: some DBs don't have native GUID support, so you will have to write some userland code to make it work.
 3. make the column non-nullable
